lua/entities/gmod_wire_expression2/core/vehicle.lua

```diff
@@ -0,0 +1,146 @@
+	exitPoint.Angle = Angle(exitAng[1], exitAng[2], 0) -- Prevent angle's roll-axis from changing.
+	VehicleExitPoints[this] = exitPoint
+end
+
+--- Sets exit position of a vehicle (returns 1 on success; otherwise, it returns 0 to indicate a failure).
+e2function number entity:setVehicleExitPos(vector exitPos)
+	if not (IsValid(this) and this:IsVehicle() and isOwner(self, this)) then return 0 end
+	-- Validate exitPos (make sure it is inside of the world).
+	local pos = Vector(exitPos[1], exitPos[2], exitPos[3])
+	--[[
+	-- I didn't like using hull-trace here..
+	local tr = util_TraceHull({
+		start = pos,
+		endpos = pos,
+		mins = Vector(-16, -16, 0),
+		maxs = Vector(16, 16, 72),
+		--mask = ...
+	})
+	if tr.Hit then return 0 end -- If the trace is hit then we quit, because the player will be stuck at exitPos.
+	]]
+	-- Let's keep it simple and effective via PointContents function:
+	if bit_band(util_PointContents(pos), CONTENTS_SOLID) == CONTENTS_SOLID then return 0 end -- Check if the contents at exitPos is "solid", we quit if it is solid, because it means that exitPos is outside of the world.
+	local exitPoint = VehicleExitPoints[this] or {}
+	exitPoint.Position = pos
+	VehicleExitPoints[this] = exitPoint
+	return 1
+end
+
+--------------------------------------------------------------------------------
+
+registerCallback("postinit", function()
+	--print("postinit")
+	hook.Add("PlayerLeaveVehicle", "E2SetVehicleExitPoint", function(ply, vehicle)
+		if not (ply:IsValid() and vehicle:IsValid()) then return end -- Don't run if the player/vehicle is not valid.
+		-- Don't run if there is no exit-point data for this vehicle.
+		local exitPoint = VehicleExitPoints[vehicle]
+		if not exitPoint then return end
+		if exitPoint.Position then -- If exit-point data has set the custom Position then "teleport" the player, taking into account for "wire_pod_exit_distance".
+			-- These four lines below are just slightly changed but still copy/paste code (from Vehicle Exit Point entity):
+			local origin = vehicle:GetPos()
+			local direction = exitPoint.Position - origin
+			local direction_distance = direction:Length()
+			ply:SetPos(origin + direction / direction_distance * math_min(direction_distance, math_max(0, wire_pod_exit_distance:GetInt())))
+		end
+		if exitPoint.Angle then -- If exit-point data has set the custom Angle then set new eye angles for the player.
+			ply:SetEyeAngles(exitPoint.Angle)
+		end
+	end)
+	hook.Add("EntityRemoved", "E2RemoveVehicleExitPoint", function(ent)
+		-- No need for redundant checks, if <ent> key exists in VehicleExitPoints table, data will be removed by setting it to nil value.
+		VehicleExitPoints[ent] = nil
+	end)
+end)
+
+--registerCallback("construct", function(...) print("construct", ...) end)
+
+-- TODO: Possibly remove exit-point data from vehicles assigned by E2 chip (when it is removed)?
+--registerCallback("destruct", function(...) print("destruct", ...) end)
```

Comment:
I think I'd prefer if the E2 destructor cleaned up the changes it made to vehicles.